src/benchmarks/micro/libraries/System.Numerics.Vectors/Perf_VectorConvert.cs

```diff
@@ -0,0 +1,232 @@
+
+        [Benchmark]
+        public Vector<double> Convert_long_double() => Convert<long, double>(s_valuesInt64);
+
+        [Benchmark]
+        public Vector<double> Convert_ulong_double() => Convert<ulong, double>(s_valuesUInt64);
+
+        // Vector.Narrow
+        [Benchmark]
+        public Vector<float> Narrow_double() => Narrow<double, float>(s_valuesDouble);
+
+        [Benchmark]
+        public Vector<sbyte> Narrow_short() => Narrow<short, sbyte>(s_valuesInt16);
+
+        [Benchmark]
+        public Vector<byte> Narrow_ushort() => Narrow<ushort, byte>(s_valuesUInt16);
+
+        [Benchmark]
+        public Vector<short> Narrow_int() => Narrow<int, short>(s_valuesInt32);
+
+        [Benchmark]
+        public Vector<ushort> Narrow_uint() => Narrow<uint, ushort>(s_valuesUInt32);
+
+        [Benchmark]
+        public Vector<int> Narrow_long() => Narrow<long, int>(s_valuesInt64);
+
+        [Benchmark]
+        public Vector<uint> Narrow_ulong() => Narrow<ulong, uint>(s_valuesUInt64);
+
+        // Vector.Widen
+        [Benchmark]
+        public Vector<double> Widen_float() => Widen<float, double>(s_valuesFloat);
+
+        [Benchmark]
+        public Vector<short> Widen_sbyte() => Widen<sbyte, short>(s_valuesSByte);
+
+        [Benchmark]
+        public Vector<ushort> Widen_byte() => Widen<byte, ushort>(s_valuesByte);
+
+        [Benchmark]
+        public Vector<int> Widen_short() => Widen<short, int>(s_valuesInt16);
+
+        [Benchmark]
+        public Vector<uint> Widen_ushort() => Widen<ushort, uint>(s_valuesUInt16);
+
+        [Benchmark]
+        public Vector<long> Widen_int() => Widen<int, long>(s_valuesInt32);
+
+        [Benchmark]
+        public Vector<ulong> Widen_uint() => Widen<uint, ulong>(s_valuesUInt32);
+
+
+        // These arrays are used for the Narrow benchmarks, so they need 2 vectors per iteration
+        private static readonly double[] s_valuesDouble = ValuesGenerator.Array<double>(Vector<double>.Count * 2 * Iterations);
+        private static readonly short[] s_valuesInt16 = ValuesGenerator.Array<short>(Vector<short>.Count * 2 * Iterations);
+        private static readonly ushort[] s_valuesUInt16 = ValuesGenerator.Array<ushort>(Vector<ushort>.Count * 2 * Iterations);
+        private static readonly int[] s_valuesInt32 = ValuesGenerator.Array<int>(Vector<int>.Count * 2 * Iterations);
+        private static readonly uint[] s_valuesUInt32 = ValuesGenerator.Array<uint>(Vector<uint>.Count * 2 * Iterations);
+        private static readonly long[] s_valuesInt64 = ValuesGenerator.Array<long>(Vector<long>.Count * 2 * Iterations);
+        private static readonly ulong[] s_valuesUInt64 = ValuesGenerator.Array<ulong>(Vector<ulong>.Count * 2 * Iterations);
+
+        // The remainder are used only for Widen and same-size Convert benchmarks
+        private static readonly float[] s_valuesFloat = ValuesGenerator.Array<float>(Vector<float>.Count * Iterations);
```

Comment:
nit: you can totally ignore it: the naming is inconsistent: sometimes you are using `C#` type names like `int` for method names, sometimes .NET type names like `Int32` for fields (but also `float` for fields, instead `single`). IMHO it would be good to choose one naming and use it everywhere